File: src/main/java/com/amir35/spring_ai/dto/response/RagResponse.java
package com.amir35.spring_ai.dto.response;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class RagResponse {

    private String question;

    private String answer;

    private List<SourceResponse> sources;

    private TokenUsageResponse tokenUsage;

    private RagPerformanceResponse performance;
}

File: spring-ai-ui/src/app/models/rag-response.ts
import { SourceResponse } from './source-response';

export interface RagResponse {

  question: string;

  answer: string;

  sources: SourceResponse[];

  tokenUsage: TokenUsageResponse;

  performance: RagPerformanceResponse;
}


export interface TokenUsageResponse {

  promptTokens: number;

  completionTokens: number;

  totalTokens: number;
}


export interface RagPerformanceResponse {

  retrievedChunks: number;

  responseTimeMs: number;

  contextTokens: number;
  chatClientTimeMs: number;
}